test/test_functional_tensor.py

```diff
@@ -868,6 +798,87 @@ def test_perspective_interpolation_warning(tester):
         tester.assertTrue(res1.equal(res2))
 
 
+@pytest.mark.parametrize('device', cpu_and_gpu())
+@pytest.mark.parametrize('dt', [None, torch.float32, torch.float64, torch.float16])
+@pytest.mark.parametrize('size', [32, 26, [32, ], [32, 32], (32, 32), [26, 35]])
+@pytest.mark.parametrize('max_size', [None, 34, 40, 1000])
+@pytest.mark.parametrize('interpolation', [BILINEAR, BICUBIC, NEAREST])
+def test_resize(device, dt, size, max_size, interpolation, tester):
+
+    torch.manual_seed(12)
+    script_fn = torch.jit.script(F.resize)
+    tensor, pil_img = tester._create_data(26, 36, device=device)
+    batch_tensors = tester._create_data_batch(16, 18, num_samples=4, device=device)
+
+    if dt == torch.float16 and device == "cpu":
+        # skip float16 on CPU case
+        return
+
+    if dt is not None:
+        # This is a trivial cast to float of uint8 data to test all cases
+        tensor = tensor.to(dt)
+        batch_tensors = batch_tensors.to(dt)
+
+    if max_size is not None and isinstance(size, Sequence) and len(size) != 1:
+        return  # unsupported, see assertRaises below
+
+    resized_tensor = F.resize(tensor, size=size, interpolation=interpolation, max_size=max_size)
+    resized_pil_img = F.resize(pil_img, size=size, interpolation=interpolation, max_size=max_size)
+
+    assert_equal(
+        resized_tensor.size()[1:],
+        resized_pil_img.size[::-1],
+    )
```

Comment:
Although `assert_equal` is able to compare sequences of scalars, I think it is overkill here. We can just rely on `pytest`'s introspection for the message:

```suggestion
    assert resized_tensor.size()[1:] == resized_pil_img.size[::-1]
```